kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/CoreUtils.java

```diff
@@ -76,4 +81,38 @@
             ));
     }
 
+    public static <T> List<T> concatList(final List<T> list1, final List<T> list2) {
+        try {
+            // if `list` can be modifiable, modify the `list1` in place.
+            list1.addAll(list2);
+            return list1;
+        } catch (UnsupportedOperationException ignored) {
```

Comment:
Personally I don't like this very much.
because it is not clear if the input list will be modified or not.

why can't we always create a new List ?

```
List<T> result = new ArrayList<>(list1.size() + list2.size());
result.addAll(list1);
result.addAll(list2);
return result;
```

it should be more efficient and less error prone.

Please note that `list1` will be resized when you add `list2` elements so in anycase you have a arraycopy behind the scenes, but you don't know, because it depends on the internals of `list1`.

Reply:
Yeah. It's right. The "in place" solution doesn't looks so good.

Reply:
I used `Streams#concat` now. See https://github.com/streamnative/kop/pull/1107/commits/b79b36a560322711754b008617256a1fec79b645.

Reply:
Why don't we use @eolivelli 's way? Look like it is more efficient.

Reply:
Good point. BTW, I found we can just use `ListUtils#union` instead of reimplementing the same logic.

Reply:
In addition, I think it's not in a critical path. The performance gap usually doesn't make a difference.